src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -785,6 +842,19 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		if (this._databaseNameInputBox) {
 			this._databaseNameInputBox.enabled = true;
 		}
+		if (this._connectionStringOptions.isEnabled) {
+			this._connectionStringInputBox.enable();
+			this._defaultInputOptionRadioButton.enabled = true;
+			this._connectionStringRadioButton.enabled = true;
+		}
+	}
+
+	public get useConnectionString(): boolean {
+		return this._connectionStringRadioButton?.checked;
+	}
+
+	public get connectionString(): string {
+		return this._connectionStringInputBox.value;
```

Comment:
you null-check or check `this._connectionStringOptions.isEnabled` before accessing properties on `this._connectionStringInputBox` in most other places.  We don't that here?

Reply:
good catch, this is not causing issue because we will always check whether connection string should be used before accessing the connectionString property, but I agree, we should have null check here as well.